authsignal/client.py

```diff
@@ -0,0 +1,256 @@
+                raise ApiException("Identify Failed", path, http_status_code=response.status_code, api_error_message=response.json()["message"])
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), path)
+    
+    def enrol_authenticator(self, user_id, authenticator_payload,  path=None):
+        """Enrols an authenticator like a phone number for SMS on behalf of the user
+        Args:
+            user_id:  A user's id. This id should be the same as the user_id used in event calls.
+            authenticator_payload:  A dictionary with the key/value of the authenticator you want to link {'oobChannel': 'SMS', 'phoneNumber': '+112345677777'}
+        """
+        _assert_non_empty_unicode(user_id, 'user_id')
+        _assert_non_empty_dict(authenticator_payload, 'authenticator_payload')
+
+        headers = {'Content-type': 'application/json',
+                    'Accept': '*/*',
+                    'User-Agent': self._user_agent()}
+        if path is None:
+            path = self._post_enorlment_url(user_id)
+        params = {}
+        timeout = self.timeout
+
+        try:
+            response = self.session.post(
+                path,
+                auth=requests.auth.HTTPBasicAuth(self.api_key, ''),
+                data=json.dumps(authenticator_payload),
+                headers=headers,
+                timeout=timeout,
+                params=params)
+            if response.status_code > 299:
+                raise ApiException("Enrol Authenticator Failed", path, http_status_code=response.status_code, api_error_message=response.json()["message"])
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), path)
+
+    def _user_agent(self):
+        return 'Authsignal Python v%s' % (self.version)
+
+    def _track_action_url(self, user_id, action_code):
+        return self.url + '/v1/users/%s/actions/%s' % (user_id,action_code)
+    
+    def _get_action_url(self, user_id, action_code, idempotency_key):
+        return self.url + '/v1/users/%s/actions/%s/%s' % (user_id, action_code, idempotency_key)
+    
+    def _get_user_url(self, user_id):
+        return self.url + '/v1/users/%s' % (user_id)
+    
+    def _post_identify_url(self, user_id):
+        return self.url + '/v1/users/%s' % (user_id)
+
+    def _post_enorlment_url(self, user_id):
+        return self.url + '/v1/users/%s/authenticators' % (user_id)
+
+class ApiException(Exception):
+    def __init__(self, message, url, http_status_code=None, body=None, api_status=None,
+                 api_error_message=None, request=None):
+        Exception.__init__(self, message)
+
+        self.url = url
+        self.http_status_code = http_status_code
+        self.body = body
+        self.api_status = api_status
+        self.api_error_message = api_error_message
+        self.request = request
```

Comment:
How does this render with your custom attributes? Think you need a `__repr__` or `__str__` (can't remember off the top of my head)